File: 排序/选择排序/selection_sort.js
function findSmallest(arr) {
  let smallest = arr[0];
  let smallest_index = 0;
  for (let index = 1; index < arr.length; index++) {
    if (arr[index] < smallest) {
      smallest = arr[index];
      smallest_index = index;
    }
  }
  return smallest_index;
}

function selectionSort(arr) {
  let newArr = [];
  while (arr.length > 0) {
    const smallest = findSmallest(arr);
    newArr.push(arr.splice(smallest, 1)[0]);
  }

  return newArr;
}

console.log("result:", selectionSort([5, 2, 7, 12, 3, 6]));

File: 排序/选择排序/selection_sort.py
def findSmallest(arr):
  smallest = arr[0]
  smallest_index = 0
  for i in range(1, len(arr)):
    if arr[i] < smallest:
      smallest = arr[i]
      smallest_index = i
  return smallest_index


def selectionSort(arr):
  newArr = []
  for _ in arr[:]:
    smallest = findSmallest(arr)
    newArr.append(arr.pop(smallest))
  return newArr

print('result: {}'.format(selectionSort([5, 2, 7, 12, 3, 6])))
